framework/src/media/utils/MediaUtils.cpp

```diff
@@ -462,5 +462,125 @@ bool header_parsing(unsigned char *buffer, unsigned int bufferSize, audio_type_t
 	}
 	return true;
 }
+
+bool write_little_endian(FILE *fp, unsigned int num, unsigned short byte)
+{
+	unsigned char buf;
+	int ret = 0;
+	for (int i = 0; i < byte; ++i) {
+		buf = num & 0xff;
+		ret += fwrite(&buf, sizeof(char), 1 ,fp);
+		num >>= 8;
+	}
+	if (ret != byte) {
+		medvdbg("file write failed error\n");
+		return false;
+	}
+	return true;
+}
+
+bool create_wav_header(FILE *fp)
+{
+	unsigned char *buffer;
+	buffer = (unsigned char *)malloc(WAVE_HEADER_LENGTH);
+	if (buffer == NULL) {
+		medvdbg("fail to malloc buffer\n");
+		if (fclose(fp) == 0) {
+			fp = nullptr;
+		}
+		return false;
+	}
+
+	memset(buffer, 0, WAVE_HEADER_LENGTH);
+	fwrite(buffer, sizeof(unsigned char), WAVE_HEADER_LENGTH, fp);
+	free(buffer);
+	return true;
+}
+
+bool write_wav_header(FILE *fp, unsigned int channel, unsigned int sampleRate, audio_format_type_t pcmFormat, unsigned int fileSize) {
+	/**
+	*wave header is
+	*Chunk ID 'RIFF' (4byte) / Chunk Size (4byte) / Fomat 'WAVE' (4byte) /
+	*Chunk ID 'fmt ' (4byte) / Chunk Size (4byte) / Audio Format (2byte) / NumChannels     (2byte) /
+	*sample Rate     (4byte) / Byte Rate  (4byte) / Block Align  (2byte) / Bits Per sample (2byte) /
+	*Chunk ID 'data' (4byte) / Chunk Size (4byte)
+	*/
+
+	unsigned int bitPerSample = 0;
+	unsigned int blockAlign = 0;
+	unsigned int byteRate = 0;
+
+	if (fseek(fp, 0, SEEK_SET) != 0) {
+		medvdbg("file seek failed error\n");
+		return false;
+	}
+
+	switch (pcmFormat) {
+	case AUDIO_FORMAT_TYPE_S8:
+		bitPerSample = 8;
+		break;
+	case AUDIO_FORMAT_TYPE_S16_LE:
+		bitPerSample = 16;
+		break;
+	case AUDIO_FORMAT_TYPE_S32_LE:
+		bitPerSample = 32;
+		break;
+	}
+
+	byteRate = sampleRate * channel * (bitPerSample >> 3);
+	blockAlign = channel * (bitPerSample >> 3);
+
+	int ret;
+
+	ret = fwrite("RIFF", sizeof(char), 4, fp);
+	if (ret != 4) {
+		medvdbg("file write failed error\n");
+		return false;
+	}
+	if (!write_little_endian(fp, fileSize - 12, 4)) {
+		return false;
+	}
+	ret = fwrite("WAVE", sizeof(char), 4, fp);
+	if (ret != 4) {
+		medvdbg("file write failed error\n");
+		return false;
+	}
+
+	ret = fwrite("fmt ", sizeof(char), 4, fp);
+	if (ret != 4) {
+		medvdbg("file write failed error\n");
+		return false;
+	}
```

Comment:
1. instead of call write_little_endian many times, you can use structure
```
struct wav_header {
    int a;
    int b;
    int c;
    int channel;
    ...
}

a = 4;
b = 2;
...
channel = 2;

 write_little_endian(fp, wav_header, sizeof(wav_header));
```

Reply:
2. if (AUDIO_FORMAT_TYPE_S16_LE || AUDIO_FORMAT_TYPE_S32_LE ) {
       write_little_endian();
    } else { 
       /* return false now but remain else for PCM_FORMAT_S16_BE or some otherthing..
      return false;
    }